aten/src/ATen/templates/RegisterBackendSelect.cpp

```diff
@@ -36,6 +36,11 @@ bool is_pinned(const Tensor& self, c10::optional<at::Device> device) {
 at::Tensor _pin_memory(const Tensor& self, c10::optional<at::Device> device) {
   TORCH_CHECK(self.device().is_cpu(), "cannot pin '", self.toString(), "' only dense CPU tensors can be pinned");
   DispatchKeySet _dk = c10::DispatchKeySet(c10::computeDispatchKey(c10::nullopt, self.layout(), device.value_or(at::kCUDA)));
+  if (self.is_nested()) {
+    auto nested_key_set = c10::DispatchKeySet(
+        {c10::DispatchKey::NestedTensor, c10::DispatchKey::AutogradNestedTensor});
+    _dk = _dk.add(self.key_set() & nested_key_set);
```

Comment:
Why add self.key_set() again?

Reply:
ah I'm trying to add only the elements of `{NestedTensor, AutogradNestedTensor}` that are present in `self.key_set()`. maybe there's a cleaner way?

Reply:
Aren't both these keys always present if `is_nested()` is true?

Reply:
Strictly speaking, `is_nested() == True` only implies that the `NestedTensor` key is present:
https://github.com/pytorch/pytorch/blob/901aa85b58e8f490631ce1db44e6555869a31893/c10/core/TensorImpl.h#L1202-L1204

I believe it's possible to have a NT with `NestedTensor` set but not `AutogradNestedTensor` (e.g. for NTs created in inference mode). cc @soulitzer to verify this

Reply:
Can confirm:
```
from torch._C import DispatchKey

a, b = torch.arange(3), torch.arange(5) + 3
nt1 = torch.nested.nested_tensor([a, b])

with torch.autograd.inference_mode():
    nt2 = torch.nested.nested_tensor([a, b])

print(torch._C._dispatch_keys(nt1).has(DispatchKey.AutogradNestedTensor))  # True
print(torch._C._dispatch_keys(nt2).has(DispatchKey.AutogradNestedTensor))  # False
```